packages/truffle-decode-utils/src/allocation.ts

```diff
@@ -107,18 +108,29 @@ export namespace Allocation {
     return { from, to, next, children: {} };
   }
 
-  function allocateDeclaration(declaration: any, refs: any[], slot: Slot, index: number): Range {
+  function allocateDeclaration(declaration: any, refs: any[], slot: Slot, index: number, allocations: any): Range {
     let definition = refs[declaration.id].definition;
     var byteSize = DefinitionUtils.storageSize(definition);  // yum
 
     if (DefinitionUtils.typeClass(definition) != "struct") {
       return allocateValue(slot, index, byteSize);
     }
+    
+    if (allocations[declaration.id] !== undefined) {
+      return allocations[declaration.id];
+    }
+
+    //NOTE: due to the way the allocator is now called, we should never reach
+    //this point in the code -- we should *only* be allocating a struct *after*
+    //all any structs it refers to have been allocated.  Nonetheless, I'm going
+    //to leave this case in for now as something of a failsafe.
+
+    //debug("Warning!  Struct failsafe invoked!")
 
     let struct = refs[definition.typeName.referencedDeclaration];
     // debug("struct: %O", struct);
 
-    let result =  allocateDeclarations(struct.variables || [], refs, slot, index);
+    let result = allocateDeclarations(struct.variables || [], refs, allocations, slot, index);
```

Comment:
I would be in favor of removing this "dead" code, and instead: reverse the `if` on L119, throwing an exception that says "Allocation not found" or somesuch. For users that have analytics enabled, the exception will get recorded there so we can see how often it happens.

Reply:
Done.